### PR TITLE
Do not set AWS_PROFILE env in generated kubeconfig

### PR DESCRIPTION
**Pull request by rquitales**
### Proposed changes

AWS_PROFILE values can vary between users despite variations providing the same level of access to the AWS resources. Always setting the profile name in the kubeconfig will mean that other users of the Pulumi program will need to ensure that their profile names also match, which isn't ideal.

This change would enable users to switch profiles, without needing to regenerate a Kubeconfig as the underlying `eks get-token` command will use the ambient `AWS_PROFILE` env var to determine the best profile.

I have manually validated that this change works locally by explicitly specifying a custom profile on the first `pulumi up`, changing the profile to another name and creating an on cluster deployment to see if the generated kubeconfig can handle the profile name change.

### Related issues (optional)

Fixes: #868

### CODE REVIEW

File: examples/utils/utils.go [hunk 1]
```diff
@@ -652,3 +660,20 @@ func clientSetFromKubeconfig(kubeconfig any) (*kubernetes.Clientset, error) {
...
+	kc, err := json.Marshal(kubeconfig)
+	if err != nil {
+		t.Errorf("unable to marshal provided kubeconfig: %s", err)
+	}
+	kubeAccess, err := KubeconfigToKubeAccess(kc)
+	if err != nil {
+		t.Errorf("unable to create KubeAccess from kubeconfig: %s", err)
+	}
+
+	_, err = kubeAccess.Clientset.Discovery().ServerVersion()
```
Comment (thomas11) [inline examples/utils/utils.go:675]: It would be good to distinguish auth errors from other errors so we don't hide those.

**Review by thomas11 — APPROVED** [review]
Sadly, the upgrade tests are failing again due to a new AMI id :(

File: examples/utils/utils.go (hunk 1 above)
Comment (rquitales) [reply examples/utils/utils.go:675]: I'm working on another PR to improve our e2e tests, so I'll roll this suggestion into that PR as well!

**Comment by fernandocarletti** [comment]
Hello! Just a suggestion for the future. This is a breaking change, since it does not offer backward compatibility with the old behavior (I know this the hard way :). It sounds this should be released as 3.0 instead as 2.3, since minor versions should keep backward compatibility.

I'm assuming Pulumi does use semver, based on [this](https://www.pulumi.com/docs/using-pulumi/pulumi-packages/schema/#:~:text=The%20version%20must%20be%20valid%20semver.), which made me overlook it before merging the dependabot PR (and yes, I know I should have checked the changelog more careful, there's a caution note there, that's why it is a friendly suggestion and not a rant hahaha).

**Comment by gunzy83** [comment]
This absolutely is a breaking change. Took me a while to track down how to deal with this on upgrade (given I am already applying [transformations](https://github.com/pulumi/pulumi-eks/issues/1038#issuecomment-2036131649) to overcome deficiencies). 

To me it makes little sense that you can create the cluster with an AWS profile (saved into the Kubeconfig for the initial deploy of the cluster resource) but then immediately creates a provider (`${name}-provider`) inside the `Cluster` resource without the profile included in its Kubeconfig (which then throws `Unable to locate credentials. You can configure credentials by running "aws configure".` even if it is not used in the program). I would assume in the same stack you would still need the profile used to create the cluster which would need to remain consistent between machines?

Even though we are using an unconventional way of accessing AWS config (using `AWS_CONFIG_FILE` to a project local dir with local and CI config files in them), I find it baffling that an org trying to deal with things at scale would at least opt for consistent profile names for environments to reduce confusion and doesn't work on my machine problems. Anyway...

**Comment by rquitales** [comment]
Thanks for your feedback on this change. We've previously received comments on the previous behavior from a number of customers and how it affects their user journeys in a multi-user setup. This coupled with the discussions in https://github.com/pulumi/pulumi-eks/issues/868 and https://github.com/pulumi/pulumi-aws/issues/1906#issuecomment-1099574436 led us to revisit the behavior of the generated kubeconfig and its behavior.
I do apologize for any pain this may have caused and I'll be sure to surface this with the rest of the team such that impacts like this are minimized in the future.

**Pull request merged**